internal/platform/database/database.go

```diff
@@ -0,0 +1,44 @@
+package database
+
+import (
+	"net/url"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq" // The database driver in use.
+)
+
+// Config is the required properties to use the database.
+type Config struct {
```

Comment:
NO! These tags are hidden. Please remove this and in the cfg struct ask for everything that is needed. Decouple this stuff.